### PR TITLE
Add a User Guide for SPMD Llama 2 Training

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request convert the user guide into a mark down and upload it to GH. The user guide is authored by Jon and me.

### CODE REVIEW

File: USER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+### Cloud TPU VM Creation
+```
+export TPU_NAME=jwtan-v4-32
+export ZONE=us-central2-b
+export ACCELERATOR_TYPE=v4-32
+export RUNTIME_VERSION=tpu-ubuntu2204-base
+export PROJECT=mlperf-high-priority-project
+
+gcloud alpha compute tpus queued-resources create $TPU_NAME --node-id $TPU_NAME --zone=$ZONE --project=$PROJECT --accelerator-type=$ACCELERATOR_TYPE --runtime-version=$RUNTIME_VERSION
```
Comment (jonb377) [inline USER_GUIDE.md]: If we split this into a few lines, there won't be a scrollbar when rendered

File: USER_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+```
+
+The last step for HF setup is to copy your Llama 2 config into the TPU VM.
+```
+gcloud compute tpus tpu-vm scp llama_2_7B.config $TPU_NAME:~/config.json --worker all --project $PROJECT --zone=$ZONE
+```
+
+
+## Steps to Run HF LLaMA
+```
```
Comment (jonb377) [inline USER_GUIDE.md]: nit: Can we add a sentence here so it doesn't jump straight from the heading to code?

File: USER_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+export TPU_NAME=jwtan-v4-32
+export ZONE=us-central2-b
+export ACCELERATOR_TYPE=v4-32
+export RUNTIME_VERSION=tpu-ubuntu2204-base
+export PROJECT=mlperf-high-priority-project
+
+gcloud alpha compute tpus queued-resources create $TPU_NAME --node-id $TPU_NAME --zone=$ZONE --project=$PROJECT --accelerator-type=$ACCELERATOR_TYPE --runtime-version=$RUNTIME_VERSION
+```
+
+Change `{TPU\_NAME, ZONE, PROJECT, RUNTIME\_VERSION}` as needed. For v5e, use the `v2-alpha-tpuv5-lite` runtime version in a supported project and zone. And use the following command to query the status of the creation request:
```
Comment (jonb377) [inline USER_GUIDE.md]: `\_` keeps the backslash when rendered.

File: USER_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+
+*   `PJRT\_DEVICE`: Specify the XLA device.
+*   `XLA\_USE\_BF16`: Force to use bfloat16 as default dtype.
+*   `XLA\_IR\_DEBUG`: Capture Python stack trace in Lazy IRs.
+*   `XLA\_HLO\_DEBUG`: Capture Python stack trace in HLOs.
+*   `PROFILE\_EPOCH`: Specify which epoch to start taking the profile.
+*   `PROFILE\_STEP`: Specify which step to start taking the profile.
+*   `PROFILE\_DURATION\_MS`: Specify how long the profiling will last.
+*   `PROFILE\_LOGDIR`: Specify where to put the profiling results.
```
Comment (jonb377) [inline USER_GUIDE.md]: Same here - the backslash shows in the result

**Review by jonb377 — APPROVED** [review]
Looks great! Just some formatting comments. What do you think about naming the file `SPMD_USER_GUIDE.md` to make it explicit?

File: USER_GUIDE.md (hunk 1 above)
Comment (alanwaketan) [reply USER_GUIDE.md]: Not sure what do you mean? Mind a screenshot. Here is my rendered version:
Screenshot 2023-09-26 at 3 38 18 PM

File: USER_GUIDE.md (hunk 3 above)
Comment (alanwaketan) [reply USER_GUIDE.md]: Good catch.

File: USER_GUIDE.md (hunk 1 above)
Comment (jonb377) [reply USER_GUIDE.md]: From https://github.com/pytorch-tpu/transformers/blob/933a18a56a612e37009ef5507bfa5bce2173fd9d/USER_GUIDE.md

image

Maybe it's just me though

File: USER_GUIDE.md (hunk 2 above)
Comment (alanwaketan) [reply USER_GUIDE.md]: Done.

File: USER_GUIDE.md (hunk 1 above)
Comment (alanwaketan) [reply USER_GUIDE.md]: Let me chop it. vscode is too smart.

**Pull request merged**